employee-rostering-frontend/src/ui/components/OptionalInput.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { Switch, TextInput } from '@patternfly/react-core';
+
+export interface OptionalInputProps<T> {
+  defaultValue: T|null;
+  isValid: (value: string) => boolean;
+  valueMapper: (value: string) => T;
+  valueToString: (value: T) => string;
+  onChange: (value: T|null) => void;
+  label: string;
+}
+
+interface OptionalInputState<T> {
+  inputValue: T|null;
+  currentValue: T|null;
+  isChecked: boolean;
```

Comment:
Can we do without the checkbox? I've tried the UI and seems to be unnecessary. I can imagine that if I delete the value from the field it's enough to make it empty. Why do I have to disable it? Or why do I have to enable it before I enter a value?

Reply:
From https://www.patternfly.org/pattern-library/forms-and-controls/data-input/: "Use an On/Off Switch when a user is presented with the binary choice to enable or disable some optional attribute or capability. " Since Maximum Daily Hours and the like are optional, it make sense to use an On/Off switch to make it clear to the user that they are, in fact, optional.

Reply:
Thanks, that's a valid argument. I'm not entirely convinced this should apply to the inline editable rows but I'm not a UX expert :slightly_smiling_face:. Let's stick with the guidelines unless UX advises otherwise.